src/main/java/org/elasticsearch/action/admin/indices/delete/TransportDeleteIndexAction.java

```diff
@@ -83,22 +85,22 @@ protected void masterOperation(final DeleteIndexRequest request, final ClusterSt
         }
         // TODO: this API should be improved, currently, if one delete index failed, we send a failure, we should send a response array that includes all the indices that were deleted
         final CountDown count = new CountDown(concreteIndices.length);
+        final AtomicReference<Throwable> lastFailure = new AtomicReference<>();
```

Comment:
Can we somehow make this a synchronized list and report all failures? I'd just add 2-N failures as suppressed to the first?